internal/controller/agenttags/tags.go

```diff
@@ -0,0 +1,72 @@
+package agenttags
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+func ToMap(tags []string) (map[string]string, []error) {
+	m := map[string]string{}
+	errs := []error{}
+	for _, tag := range tags {
+		parts := strings.SplitN(tag, "=", 2)
+		if len(parts) != 2 {
+			errs = append(errs, fmt.Errorf("invalid agent tag: %q", tag))
+			continue
+		}
+		m[parts[0]] = parts[1]
+
+	}
+	return m, errs
+}
+
+func mapToLabels(m map[string]string) (map[string]string, []error) {
+	labels := map[string]string{}
+	errs := []error{}
+	for k, v := range m {
+		namespacedKey := "buildkite.com/" + k
```

Comment:
Is it worth sub-namespacing inside "buildkite.com/"? Should namespacedKey look more like "buildkite.com/tag/key"?

Reply:
The bit after the first `/` can't contain a `/`. It has to be alphanumeric or `-` or `_`, starting and ending in an alphanumeric. So it perhaps would have to be `buildkite.com/tag-key` or `buildkite.com/tag_key`. Not a fan of either of these TBH. How about `tags.buildkite.com/key`?

Reply:
Hmm. I guess `tags.buildkite.com/key` is the best option.